container.go

```diff
@@ -20,8 +21,11 @@ type DeprecatedContainer interface {
 
 // ContainerProvider allows the creation of containers on an arbitrary system
 type ContainerProvider interface {
-	CreateContainer(context.Context, ContainerRequest) (Container, error) // create a container without starting it
-	RunContainer(context.Context, ContainerRequest) (Container, error)    // create a container and start it
+	CreateContainer(context.Context, ContainerRequest) (Container, error)      // create a container without starting it
+	CreateNetwork(context.Context, NetworkRequest) (Network, error)            // create a network
```

Comment:
I would be better to treat networks as a domain object on their own and not mix with the containers. 

You can check Java's implementation for the inspiration:
https://www.testcontainers.org/features/networking/#advanced-networking

Reply:
Yeah, I created it in the ContainerProvider because it seemed to me like Docker was the provider, and in that sense it would also be the network provider.

I can rework it to have a NetworkProvider 👍

Reply:
You're right: I will try to mimic the Java implementation as much as possible, thanks!

Reply:
Addresses here: https://github.com/testcontainers/testcontainers-go/pull/76/commits/1e5969298fbba8915301b38e3ef3f7616a62559d